persistent-2.11.0.3 can't build

Can't build with ghc-8.10.4 and ghc-9.0.1

culprit https://github.com/yesodweb/persistent/commit/e7a7842670b48d84453afc9c2b124d66e6dcf83d

```
[  107s] [12 of 24] Compiling Database.Persist.Sql.Types.Internal ( Database/Persist/Sql/Types/Internal.hs, dist/build/Database/Persist/Sql/Types/Internal.o, dist/build/Database/Persist/Sql/Types/Internal.dyn_o )
[  107s] 
[  107s] Database/Persist/Sql/Types/Internal.hs:34:1: warning: [-Wunused-imports]
[  107s]     The import of ‘Data.Monoid’ is redundant
[  107s]       except perhaps to import instances from ‘Data.Monoid’
[  107s]     To import instances alone, use: import Data.Monoid()
[  107s]    |
[  107s] 34 | import Data.Monoid ((<>))
[  107s]    | ^^^^^^^^^^^^^^^^^^^^^^^^^
[  108s] [13 of 24] Compiling Database.Persist.Sql.Types ( Database/Persist/Sql/Types.hs, dist/build/Database/Persist/Sql/Types.o, dist/build/Database/Persist/Sql/Types.dyn_o )
[  108s] 
[  108s] Database/Persist/Sql/Types.hs:11:1: warning: [-Wunused-imports]
[  108s]     The import of ‘Database.Persist.Types.Base’ is redundant
[  108s]       except perhaps to import instances from ‘Database.Persist.Types.Base’
[  108s]     To import instances alone, use: import Database.Persist.Types.Base()
[  108s]    |
[  108s] 11 | import Database.Persist.Types.Base (FieldCascade)
[  108s]    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[  109s] [14 of 24] Compiling Database.Persist.Sql.Internal ( Database/Persist/Sql/Internal.hs, dist/build/Database/Persist/Sql/Internal.o, dist/build/Database/Persist/Sql/Internal.dyn_o )
[  109s] [15 of 24] Compiling Database.Persist ( Database/Persist.hs, dist/build/Database/Persist.o, dist/build/Database/Persist.dyn_o )
[  110s] [16 of 24] Compiling Database.Persist.Sql.Util ( Database/Persist/Sql/Util.hs, dist/build/Database/Persist/Sql/Util.o, dist/build/Database/Persist/Sql/Util.dyn_o )
[  110s] 
[  110s] Database/Persist/Sql/Util.hs:24:1: warning: [-Wunused-imports]
[  110s]     The import of ‘Data.Monoid’ is redundant
[  110s]       except perhaps to import instances from ‘Data.Monoid’
[  110s]     To import instances alone, use: import Data.Monoid()
[  110s]    |
[  110s] 24 | import Data.Monoid ((<>))
[  110s]    | ^^^^^^^^^^^^^^^^^^^^^^^^^
[  110s] [17 of 24] Compiling Database.Persist.Sql.Class ( Database/Persist/Sql/Class.hs, dist/build/Database/Persist/Sql/Class.o, dist/build/Database/Persist/Sql/Class.dyn_o )
[  112s] 
[  112s] Database/Persist/Sql/Class.hs:24:1: warning: [-Wunused-imports]
[  112s]     The import of ‘Data.Monoid’ is redundant
[  112s]       except perhaps to import instances from ‘Data.Monoid’
[  112s]     To import instances alone, use: import Data.Monoid()
[  112s]    |
[  112s] 24 | import Data.Monoid ((<>))
[  112s]    | ^^^^^^^^^^^^^^^^^^^^^^^^^
[  135s] [18 of 24] Compiling Database.Persist.Sql.Raw ( Database/Persist/Sql/Raw.hs, dist/build/Database/Persist/Sql/Raw.o, dist/build/Database/Persist/Sql/Raw.dyn_o )
[  136s] [19 of 24] Compiling Database.Persist.Sql.Run ( Database/Persist/Sql/Run.hs, dist/build/Database/Persist/Sql/Run.o, dist/build/Database/Persist/Sql/Run.dyn_o )
[  136s] 
[  136s] Database/Persist/Sql/Run.hs:99:9: error: parse error on input ‘let’
[  136s]    |
[  136s] 99 |         let getter = getStmtConn sqlBackend
[  136s]    |         ^^^
```

--- a/persistent/ChangeLog.md
+++ b/persistent/ChangeLog.md
@@ -1,5 +1,10 @@
 # Changelog for persistent
 
+## 2.11.0.4
+
+* Fix a compile error [#1213](https://github.com/yesodweb/persistent/pull/1213)
+    * That's what I get for running this stuff with flakey CI!
+
 ## 2.11.0.3
 
 * Backported the fix from [#1207](https://github.com/yesodweb/persistent/pull/1207) for asynchronous exceptions.

--- a/persistent/Database/Persist/Quasi.hs
+++ b/persistent/Database/Persist/Quasi.hs
@@ -1200,14 +1200,14 @@ parseCascade allTokens =
                             Nothing ->
                                 go (this : acc) mupd mdel rest
     nope msg =
-        error $ msg <> ", tokens: " <> show allTokens
+        error $ msg `mappend` ", tokens: " `mappend` show allTokens
 
 parseCascadeAction
     :: CascadePrefix
     -> Text
     -> Maybe CascadeAction
 parseCascadeAction prfx text = do
-    cascadeStr <- T.stripPrefix ("On" <> toPrefix prfx) text
+    cascadeStr <- T.stripPrefix ("On" `mappend` toPrefix prfx) text
     case readEither (T.unpack cascadeStr) of
         Right a ->
             Just a

--- a/persistent/Database/Persist/Sql/Run.hs
+++ b/persistent/Database/Persist/Sql/Run.hs
@@ -94,7 +94,7 @@ runSqlPool
 runSqlPool r pconn =
     withRunInIO $ \runInIO ->
     withResource pconn $ \conn ->
-    UE.mask $ \restore ->
+    UE.mask $ \restore -> do
         let sqlBackend = projectBackend conn
         let getter = getStmtConn sqlBackend
         restore $ connBegin sqlBackend getter Nothing

--- a/persistent/persistent.cabal
+++ b/persistent/persistent.cabal
@@ -1,5 +1,5 @@
 name:            persistent
-version:         2.11.0.3
+version:         2.11.0.4
 license:         MIT
 license-file:    LICENSE
 author:          Michael Snoyman <michael@snoyman.com>

--- a/persistent/test/main.hs
+++ b/persistent/test/main.hs
@@ -9,6 +9,7 @@ import qualified Data.Map as Map
 import Data.Time
 import Text.Shakespeare.Text
 import Data.List
+import Data.Monoid
 
 import Database.Persist.Class.PersistField
 import Database.Persist.Quasi

